assets/src/edit-story/components/errorBoundary/index.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+/**
+ * Internal dependencies
+ */
+import ErrorActions from './errorActions';
+
+class ErrorBoundary extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+  };
+
+  state = {
+    hasError: false,
+    counter: 0,
+  };
+
+  reRender = () => {
+    this.setState((state) => ({
+      ...state,
+      counter: state.counter + 1,
+      hasError: false,
+    }));
+  };
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  onError(evt) {
+    console.log('onerror', evt, evt.error instanceof TypeError);
+  }
+
+  onUnhandledRejection(evt) {
+    // Should not crash the editor, more for developers, so we can remove it from here
+    console.log('onunhandledrejection', evt);
+  }
+
+  componentDidMount() {
+    window.addEventListener('error', this.onError);
```

Comment:
What are these for?

Reply:
Just for testing other kinds of errors (other than React tree error). I feel like we don't have to do much with them (in this first pass) - they will "silently fail" by default.

Reply:
If we plan to deploy error boundary in multiple locations in the component tree, then there's no way of knowing that an unhandled error came from one or another part of the component tree - it's just a bunch of handlers that will be listening on unhandled errors w/o being able to attribute them to anything.

Reply:
I was thinking about plain error reporting (to 3P service) originally, but in that case (multiple error boundaries) it will be better to create one separate component just for that.

Reply:
Yep, I think so as well.